server/util/collections.py

```diff
@@ -0,0 +1,22 @@
+# pylint: disable=invalid-name
+from typing import Optional, TypeVar, Iterable
+import itertools
+
+T = TypeVar("T")
+
+
+def group_by_iter(items: Iterable[T], key=None):
+    return itertools.groupby(sorted(items, key=key), key=key)
+
+
+def group_by(items: Iterable[T], key=None):
```

Comment:
I know this code was already written before this PR, but a quick comment with some sample inputs and outputs could be helpful. Even pointing readers to the test (which has sample inputs and outputs) could be sufficient.

I also wonder if we should rename `key` `key_fn` for clarity, since it's a function

Reply:
Thanks, I'll add some documentation. `key` is the Python convention (see `sorted` and `itertools.groupby` that are used in this file), so I'll probably leave that